### DISCUSSION TITLE
Could we support Redis Cluster?

### DISCUSSION THREAD

**Discussion by wy96f** [langfuse/langfuse#4691] [Ideas]
### Describe the feature or potential improvement

In our production environment, we only support Redis Cluster, but Langfuse currently does not support Redis Cluster. I have found that Langfuse uses BullMQ, and BullMQ can support Redis Cluster(It seems not complex?). Could we support Redis Cluster?

### Additional information

https://github.com/taskforcesh/bullmq/blob/3b218ff3a3af4286068b7aaee8f0d0909fd4f52e/docs/gitbook/patterns/redis-cluster.md?plain=1#L3
https://docs.bullmq.io/bull/patterns/redis-cluster
FYI

**Comment by Steffen911** [comment]
Hey @wy96f,
Thank you for the suggestion. Would you be willing to contribute to add this feature? Do you expect any incompatibility when making those prefix changes for non-clustered mode?

**Reply by wy96f** [reply]
@Steffen911 I've never written TypeScript before, so this might be a bit challenging for me :(. Considering compatibility, Redis configuration can be set up with multiple nodes (which can later be compatible with sentinel and cluster modes); when initializing the queue, we can add an environment variable (whether it's Redis cluster mode) to control whether to add a prefix as a cluster "hash tag". This change would make the extension better.

**Reply by marcklingen** [reply]
Open for contributions on this though as this might be a nice extension of Langfuse redis support.

**Comment by Steffen911** [comment]
We'll add Redis Cluster support via https://github.com/langfuse/langfuse/pull/7162. This should be included in one of the next OSS releases.
